Flaky unit test: TestLoadImageTimeoutError

### Description
The test misses a mock docker client call and is flaky due to this.

<!-- Not required for feature requests -->
### Expected Behavior
test passes

### Observed Behavior
test fails

### Supporting Log Snippets
```
--- FAIL: TestLoadImageTimeoutError (0.00s)

	<autogenerated>:1: missing call(s) to *mock_dockeriface.MockClient.LoadImage(is anything)

	<autogenerated>:1: aborting test due to missing call(s)
 
```

--- a/agent/engine/docker_client_test.go
+++ b/agent/engine/docker_client_test.go
@@ -37,6 +37,7 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/utils/ttime/mocks"
 
 	"context"
+
 	"github.com/aws/aws-sdk-go/aws"
 	docker "github.com/fsouza/go-dockerclient"
 	"github.com/golang/mock/gomock"
@@ -1021,7 +1022,7 @@ func TestLoadImageTimeoutError(t *testing.T) {
 	wait.Add(1)
 	mockDocker.EXPECT().LoadImage(gomock.Any()).Do(func(x interface{}) {
 		wait.Wait()
-	})
+	}).MaxTimes(1)
 
 	err := client.LoadImage(nil, time.Millisecond)
 	assert.Error(t, err)

